src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/nowhitespaceafter/InputNoWhitespaceAfterSynchronized.java

```diff
@@ -0,0 +1,55 @@
+/*
+NoWhitespaceAfter
+allowLineBreaks = (default)true
+tokens = LITERAL_SYNCHRONIZED
+
+
+*/
+package com.puppycrawl.tools.checkstyle.checks.whitespace.nowhitespaceafter;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+public class InputNoWhitespaceAfterSynchronized<T> {
+
+    private final Object lock = new Object();
+
+    void m1() {
+        synchronized (lock) {} // violation
+        synchronized(lock) {} // ok
+    }
+
+    synchronized int m2() { // ok
+        return 2;
+    }
+
+    synchronized private int[] m3() { // ok
+        return new int[]{2};
+    }
+
+    synchronized int[] m4() { // ok
+        return new int[]{2};
+    }
+
+    private final synchronized int m5() { // ok
+        return 2;
+    }
+
+    synchronized T m6() { // ok
+        return null;
+    }
+
+    @SyncAnno synchronized T m7() { // ok
+        return null;
+    }
+
+    @SyncAnno private synchronized T m8() { // ok
+        return null;
+    }
+
+    synchronized <T2> void m9() {}
```

Comment:
it looks like this case 
`synchronized<T> void method() {}` should report a violation, but it does not (?)

Reply:
I disagree. In this context, `synchronized` is a modifier, not part of a statement.

Take a look at all of the tokens available to this check:
```
            TokenTypes.ARRAY_INIT,
            TokenTypes.AT,
            TokenTypes.INC,
            TokenTypes.DEC,
            TokenTypes.UNARY_MINUS,
            TokenTypes.UNARY_PLUS,
            TokenTypes.BNOT,
            TokenTypes.LNOT,
            TokenTypes.DOT,
            TokenTypes.TYPECAST,
            TokenTypes.ARRAY_DECLARATOR,
            TokenTypes.INDEX_OP,
            TokenTypes.LITERAL_SYNCHRONIZED,
            TokenTypes.METHOD_REF,
```

`synchronized` already looks out of place, and we can see that there is no modifiers at all in this collection of tokens. So I would think that this check should not validate "no whitespace after" `synchronized` as a modifier at all.

If we did, then we would need to consider adding support for all modifiers to this check, since according to your example (`synchronized<T> void method() {}`), the following should all be violations, too:
```
public class Test {
    synchronized private static<T> void m1(){}
    synchronized private<T> void m2(){}
    private<T> void m3(){}
    strictfp<T> void m4(){}
}
```
Typically, only generic types have the type parameter immediately follow the type name; generic method type parameters are always surrounded by whitespace.

Take the openjdk19u repo as an example:
```
➜  jdk19u git:(master) find . -name "*.java" | wc -l
47947
➜  jdk19u git:(master) ag " [a-z]+<[A-Z]>" --java | wc -l 
113
➜  jdk19u git:(master) ag " [a-z]+ <[A-Z]>" --java | wc -l
2441


```

Surrounding type parameters on generic methods by whitespace is about 22 times more prevalent than not.

Reply:
From https://checkstyle.sourceforge.io/config_whitespace.html#GenericWhitespace:

>Left angle bracket ("<"):

> - should be preceded with whitespace only in generic methods definitions.
> - should not be preceded with whitespace when it is preceded method name or constructor.
> - should not be preceded with whitespace when following type name.
> - should not be followed with whitespace in all cases.

Reply:
I feel this should be moved to a new issue and not part of this as it is a bug fix, especially if the token wasn't reported before.

I truthfully can't see someone wanting this option and it is probably better to wait for a request.

Reply:
ok, I agree

Reply:
I found this also very confusing.
Please add above/below new test code
`synchronized<T2> void m9NoSpace() {} // ok, as we ignore tokens that are used in modifiers`

To make it clear that we noticed it and do this on purpose

Reply:
Done